src/api.cc

```diff
@@ -167,9 +149,55 @@ CheckInResult AkliteClient::CheckIn() const {
       }
     }
   }
-
   std::sort(targets.begin(), targets.end(), compareTargets);
-  return CheckInResult(status, client_->config.provision.primary_ecu_hardware_id, targets);
+  return targets;
+}
+
+CheckInResult AkliteClient::CheckIn() const {
+  if (!configUploaded_) {
+    client_->reportAktualizrConfiguration();
+    configUploaded_ = true;
+  }
+  client_->reportNetworkInfo();
+  client_->reportHwInfo();
+  client_->reportAppsState();
+
+  auto status = CheckInResult::Status::Ok;
+  Uptane::HardwareIdentifier hwidToFind(client_->config.provision.primary_ecu_hardware_id);
+
+  LOG_INFO << "Refreshing Targets metadata";
+  const auto rc = client_->updateImageMeta();
```

Comment:
I think we can move this online checkin to the new TUF api usage too.
Also, it looks like we need to add another method to the `TufRepo` interface - `checkMeta` which can be used instead of `client_->checkImageMetaOffline`.

Reply:
I'll leave this changes to a new commit, yet to be created

Reply:
Makes sense.